data-prepper-core/src/test/java/org/opensearch/dataprepper/pipeline/router/RouterTest.java

```diff
@@ -190,9 +196,45 @@ void route_with_one_DataFlowComponents_And_PipelineConnector() {
             final DataFlowComponent<PipelineConnector> dataFlowComponent = mock(DataFlowComponent.class);
             pipelineDataFlowComponents.add(dataFlowComponent);
 
-            createObjectUnderTest().route(recordsIn, pipelineDataFlowComponents, pipelineComponentRecordsConsumer);
+            createObjectUnderTest().route(recordsIn, pipelineDataFlowComponents, null, pipelineComponentRecordsConsumer);
 
             verify(dataFlowComponentRouter).route(recordsIn, dataFlowComponent, recordsToRoutes, null, pipelineComponentRecordsConsumer);
         }
+
+        @Test
+        void route_with_multiple_DataFlowComponents_And_Strategy() {
+            final DataFlowComponent dataFlowComponent = mock(DataFlowComponent.class);
+            dataFlowComponents = new ArrayList<>();
+            for (int i = 0; i < 5; i++) {
+                dataFlowComponents.add(dataFlowComponent);
+            }
+            RouterCopyRecordStrategy routerCopyRecordStrategy = new RouterCopyRecordStrategy(dataFlowComponents);
```

Comment:
We want to remove these tests, but this is not a unit test - you are testing two classes now.

Reply:
If you want to cover that the copy works as expected, then this is an integration test. Please use the somewhat new integration test framework.

Here is a sample integration test:
https://github.com/opensearch-project/data-prepper/blob/main/data-prepper-core/src/integrationTest/java/org/opensearch/dataprepper/integration/Router_SingleRouteIT.java

I'd say this is optional.